internal/config/global_config.go

```diff
@@ -1,80 +1,97 @@
-		MaxIdleTime time.Duration `env:"MAX_IDLETIME" envDefault:"5m"`
-		MaxLifetime time.Duration `env:"MAX_LIFETIME" envDefault:"10m"`
-		LogLevel    int           `env:"LOG_LEVEL" envDefault:"1"`
-	} `envPrefix:"DB_"`
+		Host        string
+		Port        uint16
+		Name        string
+		User        string
+		Password    string
+		MaxIdleTime time.Duration
+		MaxLifetime time.Duration
+		MaxOpenConn int
+		MaxIdleConn int
+		LogLevel    int
+	}
 	Logging struct {
-		Level  int  `env:"LEVEL" envDefault:"1"`
-		Stdout bool `env:"STDOUT" envDefault:"false"`
-	} `envPrefix:"LOGGING_"`
+		Level  int
+		Stdout bool
+	}
 	Cloudwatch struct {
-		Enabled bool   `env:"ENABLED" envDefault:"false"`
-		Group   string `env:"GROUP"`
-		Stream  string `env:"STREAM"`
-	} `envPrefix:"CLOUDWATCH_"`
+		Enabled bool
+		Group   string
+		Stream  string
+	}
 	AWS struct {
-		Region  string `env:"REGION"`
-		Key     string `env:"KEY,unset"`
-		Secret  string `env:"SECRET,unset"`
-		Session string `env:"SESSION,unset"`
-	} `envPrefix:"AWS_"`
+		Region  string
+		Key     string
+		Secret  string
+		Session string
+	}
 	FeatureFlags struct {
-		Environment string `env:"ENVIRONMENT,required" envDefault:"development"`
-	} `envPrefix:"FF_"`
-	ExitOnPanic bool `env:"EXIT_ON_PANIC" envDefault:"false"`
+		Environment string
+	}
+	ExitOnPanic bool
 }
 
 var config GlobalConfig
-var configInitialized bool
 var configMutex sync.Mutex
 
-func initializeConfig() {
-	// read missing default environmental values from .env.default
-	err := godotenv.Load(".env.default")
-	if err != nil {
-		panic(err)
-	}
+func init() {
+	var err error
 
-	// read user-defined overrides from .env
-	err = godotenv.Overload()
-	if err != nil {
-		panic(err)
+	viper.SetDefault("logging.level", 1)
+	if clowder.IsClowderEnabled() {
+		fmt.Println("Loading Clowder configuration")
+		cfg := clowder.LoadedConfig
+
+		viper.SetDefault("database.host", cfg.Database.Hostname)
+		viper.SetDefault("database.port", cfg.Database.Port)
+		viper.SetDefault("database.user", cfg.Database.Username)
+		viper.SetDefault("database.password", cfg.Database.Password)
+		viper.SetDefault("database.name", cfg.Database.Name)
+	} else {
+		fmt.Println("Loading local configuration")
+
+		viper.SetDefault("database.host", "localhost")
+		viper.SetDefault("database.port", 5432)
+		// read missing default environmental values from .env.default
+		viper.SetConfigFile(".env.default")
+		err = viper.ReadInConfig()
+		if err != nil {
+			panic(err)
+		}
+		viper.SetConfigFile(".env")
+		err = viper.ReadInConfig()
+		if err != nil {
+			panic(err)
+		}
+
+		viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+		viper.AutomaticEnv()
 	}
 
-	// parse variables into the Go type
-	err = env.Parse(&config)
+	err = viper.Unmarshal(&config)
 	if err != nil {
 		panic(err)
 	}
-	configInitialized = true
 }
 
 func GetLoggingConfig() *GlobalConfig {
 	configMutex.Lock()
 	defer configMutex.Unlock()
```

Comment:
Remove mutex completely, function `init` is guaranteed to be called during package importing (very early in program loading: https://www.developer.com/languages/inti-function-golang/

I like this approach much better, this looks like a great fit for config loading.